cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -420,6 +430,28 @@ traceD tr meta msg d = traceNamedObject tr (meta, LogValue msg (PureD d))
 traceI :: Integral i => Trace IO a -> LOMeta -> Text -> i -> IO ()
 traceI tr meta msg i = traceNamedObject tr (meta, LogValue msg (PureI (fromIntegral i)))
 
+sendEKGDirectInt :: Integral a => EKGDirect -> Text -> a -> IO ()
+sendEKGDirectInt ekgDirect name val = do
+ registeredMap <- readIORef (ekgGauges ekgDirect)
+ case SMap.lookup name registeredMap of
+   Just gauge -> Gauge.set gauge (fromIntegral val)
+   Nothing -> do
+     gauge <- EKG.getGauge name (ekgServer ekgDirect)
+     let registeredMap' = SMap.insert name gauge registeredMap
+     writeIORef (ekgGauges ekgDirect) registeredMap'
+     Gauge.set gauge (fromIntegral val)
```

Comment:
What happens if this is run from multiple threads and `getGauge` gets called multiple times for the same name?